com.woltlab.wcf/option.xml

```diff
@@ -970,6 +973,39 @@ redis:wcf.acp.option.cache_source_type.redis</selectoptions>
 				<minvalue>96</minvalue>
 				<suffix>pixel</suffix>
 			</option>
+			<option name="attachment_image_autoscale">
+				<categoryname>message.attachment.autoscale</categoryname>
+				<optiontype>boolean</optiontype>
+				<defaultvalue>1</defaultvalue>
```

Comment:
How about 1920×1200? That covers the largest typical desktop resolution and everything above gets scaled down.

Reply:
My though for defaulting to square dimensions was to accommodate smartphone screenshots, which usually come in portrait orientation but scaling them to 675×1200 in case of Full HD smartphone panels should still ok, I guess.

Reply:
I thought about this too, but smartphone differ greatly in size and there really isn't a common ground. I have a few ideas about auto-scaling images from 4K screens, but in the end they don't end up well.

Recently, I dug through the various screen resolutions of handheld devices and I cam up with this range of screen dimensions. The values denote the range of sizes for the *sm*allest and the *w*i*d*est side regardless of the orientation. They're not perfect, but they do give an idea of the ranges to expect (notably, there are actually no overlaps!).

*iPhone*
sm: 320-414
wd: 568-896

*iPad*
sm: 768
wd: 1024

*Android Phone*
sm: 360-480
wd: 640-853

*Android Tablet*
sm: 600-850
wd: 960-1024